src/runtime_src/xocl/api/clEnqueueNDRangeKernel.cpp

```diff
@@ -420,9 +420,6 @@ clEnqueueNDRangeKernel(cl_command_queue command_queue,
   xocl::enqueue::set_event_action(umEvent.get(),xocl::enqueue::action_ndrange_migrate,mEvent,kernel);
   xocl::profile::set_event_action(umEvent.get(),xocl::profile::action_ndrange_migrate,mEvent,kernel);
   xocl::appdebug::set_event_action(umEvent.get(),xocl::appdebug::action_ndrange_migrate,mEvent,kernel);
```

Comment:
this event also needs a callback to record ndrange migrate right before kernel starts

Reply:
In case the buffer is already resident, this event does nothing. lop callback can check that condition and not log it. however, if the buffer isn't resident, this actually does a buffer transfer.

Reply:
Added callback to catch buffer writes due to ndrange migration in latest commit